root/components/Tags.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {l} from '../static/scripts/common/i18n';
+import TagLink from '../static/scripts/common/components/TagLink';
+import loopParity from '../utility/loopParity';
+
+type Props = {|
+  +displayGenres: $ReadOnlyArray<AggregatedTagT>,
+  +displayTags: $ReadOnlyArray<AggregatedTagT>,
+|};
+
+const TagsComponent = ({
+  displayGenres,
+  displayTags,
+}: Props) => {
+  return (
+
+    <div id="all-tags">
+      {(displayGenres && displayGenres.length) ||
+      (displayTags && displayTags.length)
+        ? (
+          <>
+            <h2>{l('Genres')}</h2>
+
+            <div id="genres">
+              {displayGenres && displayGenres.length ? (
+                <ul className="genre-list">
+                  {displayGenres.map((genre, index) => (
+                    <li className={loopParity(index)} key={genre.tag}>
+                      <TagLink tag={genre.tag} />
+                      <span className="tag-vote-buttons">
+                        <span className="tag-count">{genre.count}</span>
+                      </span>
+                    </li>
+                  ))}
+                </ul>
+              ) : (
+                <p>{l('There are no genres to show.')}</p>
+              )}
+            </div>
+
+            <h2>{l('Other tags')}</h2>
+
+            <div id="tags">
+              {displayTags && displayTags.length ? (
+                <ul className="tag-list">
+                  {displayTags.map((tag, index) => (
+                    <li className={loopParity(index)} key={tag.tag}>
+                      <TagLink tag={tag.tag} />
+                      <span className="tag-vote-buttons">
+                        <span className="tag-count">{tag.count}</span>
+                      </span>
+                    </li>
+                  ))}
+                </ul>
+              ) : (
+                <p>{l('There are no other tags to show.')}</p>
+              )}
+            </div>
+          </>
+        ) : null
+      }
+    </div>
+  );
+};
+
+export default TagsComponent;
```

Comment:
Do you remember what the issue was with just modifying `MainTagEditor` to add `$c.user_exist` to hide the login-only parts? It doesn't seem too difficult at a glance, so I'd prefer we did that instead of duplicating the component.

Reply:
IIRC it was that the entire logic of the component was based on users existing and on voting, so it was hard to just hide the login-only parts.